src/clientdynamicinfo.cpp

```diff
@@ -37,19 +37,25 @@ ClientDynamicInfo ClientDynamicInfo::getCurrent()
 
     return {
         screen_size, real_gui_scaling, real_hud_scaling,
-        ClientDynamicInfo::calculateMaxFSSize(screen_size, gui_scaling),
+        ClientDynamicInfo::calculateMaxFSSize(screen_size, density, gui_scaling),
         touch_controls
     };
 }
 
-v2f32 ClientDynamicInfo::calculateMaxFSSize(v2u32 render_target_size, f32 gui_scaling)
+v2f32 ClientDynamicInfo::calculateMaxFSSize(v2u32 render_target_size, f32 density, f32 gui_scaling)
 {
-    f32 factor = (g_settings->getBool("touch_gui") ? 10 : 15) / gui_scaling;
-    f32 ratio = (f32)render_target_size.X / (f32)render_target_size.Y;
-    if (ratio < 1)
-        return { factor, factor / ratio };
-    else
-        return { factor * ratio, factor };
+	// must stay in sync with GUIFormSpecMenu::calculateImgsize
```

Comment:
Is it perhaps possible to make a static helper function in `GUIFormSpecMenu` that is called both here and in `calculateImgsize`?  Though it might be cleaner as it is given all the extra code in `calculateImgsize`

Reply:
Here's some refactoring: e786862fc6953f3d54f3717af9e32d1764465b5e